File: software/examples/rust/example_threshold.rs
use std::{error::Error, io, thread};
use tinkerforge::{barometer_v2_bricklet::*, ipconnection::IpConnection};

const HOST: &str = "127.0.0.1";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Barometer Bricklet 2.0

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection
    let barometer_v2_bricklet = BarometerV2Bricklet::new(UID, &ipcon); // Create device object

    ipcon.connect(HOST, PORT).recv()??; // Connect to brickd
                                        // Don't use device before ipcon is connected

    //Create listener for air pressure events.
    let air_pressure_listener = barometer_v2_bricklet.get_air_pressure_receiver();
    // Spawn thread to handle received events. This thread ends when the barometer_v2_bricklet
    // is dropped, so there is no need for manual cleanup.
    thread::spawn(move || {
        for event in air_pressure_listener {
            println!("Air Pressure: {}{}", event as f32 / 1000.0, " mbar");
            println!("Enjoy the potentially good weather!");
        }
    });

    // Configure threshold for air pressure "greater than 1025 mbar"
    // with a debounce period of 1s (1000ms)
    barometer_v2_bricklet.set_air_pressure_callback_configuration(1000, false, '>', 1025 * 1000, 0);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}

File: software/examples/go/example_threshold.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/barometer_v2_bricklet"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Barometer Bricklet 2.0.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	b, _ := barometer_v2_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	b.RegisterAirPressureCallback(func(airPressure int32) {
		fmt.Printf("Air Pressure: %f hPa\n", float64(airPressure)/1000.0)
		fmt.Println("Enjoy the potentially good weather!")
	})

	// Configure threshold for air pressure "greater than 1025 hPa"
	// with a debounce period of 1s (1000ms).
	b.SetAirPressureCallbackConfiguration(1000, false, '>', 1025*1000, 0)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()
}
